code/script.js

```diff
@@ -187,29 +198,248 @@ const CHARACTERS = [
+    eyes: 'blue',
+    color: 'light',
+    accessories: ['a lightsaber'],
+    other: ['jedi', 'sithlord'],
+    species: 'human'
+  },
+  {
+    name: 'Maul',
+    img: 'images/maul.jfif',
+    hair: 'no',
+    eyes: 'red',
+    color: 'red',
+    accessories: ['horns'],
+    other: ['sithlord'],
+    species: 'zabrak'
+  },
+  {
+    name: 'Luke Skywalker',
+    img: 'images/luke.jfif',
+    hair: 'brown',
+    eyes: 'blue',
+    color: 'light',
+    accessories: ['a lightsaber'],
+    other: ['jedi', 'rebel'],
+    species: 'human'
+  },
+  {
+    name: 'Grogu',
+    img: 'images/grogu.jfif',
+    hair: 'white',
+    eyes: 'brown',
+    color: 'green',
+    accessories: [],
+    other: [],
+    species: 'unknown'
+  },
+  {
+    name: 'Cassian',
+    img: 'images/cassian.jfif',
+    hair: 'brown',
+    eyes: 'brown',
+    color: 'light',
+    accessories: [],
+    other: ['rebel'],
+    species: 'human'
+  },
+  {
+    name: 'Leia',
+    img: 'images/leia.jfif',
+    hair: 'brown',
+    eyes: 'brown',
+    color: 'light',
+    accessories: ['a blaster', 'hairbuns'],
+    other: ['rebel'],
+    species: 'human'
+  },
+  {
+    name: 'C-3PO',
+    img: 'images/c-3po.jfif',
+    hair: 'no',
+    eyes: 'yellow',
+    color: 'yellow',
+    accessories: [],
+    other: ['protocol droid'],
+    species: 'droid'
+  },
+  {
+    name: 'Chewbacca',
+    img: 'images/chewbacca.jfif',
+    hair: 'brown',
+    eyes: 'blue',
+    color: 'brown',
+    accessories: [],
+    other: ['rebel'],
+    species: 'wookie'
+  },
+  {
+    name: 'Han Solo',
+    img: 'images/han-solo.jfif',
+    hair: 'brown',
+    eyes: 'green',
+    color: 'light',
+    accessories: ['a blaster'],
+    other: ['rebel'],
+    species: 'human'
+  },
+  {
+    name: 'Thrawn',
+    img: 'images/thrawn.jfif',
+    hair: 'blue',
+    eyes: 'red',
+    color: 'blue',
+    accessories: [],
+    other: ['grand admiral'],
+    species: 'chiss'
+  },
+  {
+    name: 'R2-D2',
+    img: 'images/r2-d2.jfif',
+    hair: 'no',
+    eyes: 'no',
+    color: 'white',
+    accessories: [],
+    other: ['astromech droid'],
+    species: 'droid'
+  },
+  {
+    name: 'Yoda',
+    img: 'images/yoda.jfif',
+    hair: 'white',
+    eyes: 'brown',
+    color: 'green',
+    accessories: [],
+    other: ['jedi'],
+    species: 'unknown'
+  }
+]
+
 // Global variables
 let secret
-let currentQuestion
+let currentQuestions
 let charactersInPlay
+let numberOfQuestions
+let startTime
+let elapsedTime
+
+//add the categories and values from the choosen characters at the beginning of the game
+const generateQuestions = () => {
+   let allCharacteristics = {}
+
+  const theme = themes.options[themes.selectedIndex].label
+
+  if(theme === 'Star Wars'){
+    charactersInPlay = CHARACTERS2
+  }
+
+  else{
+    charactersInPlay = CHARACTERS
+  }
+
+  charactersInPlay.forEach((person) => {
+    //for every key in every person, check if it exist as a property in the object allCharacteristics
+    for (let key in person) {
```

Comment:
Is this for sorting out characters that isn't the secret one?

Reply:
Realized the filter function was further down. Maybe this could be explained a little bit closer?